server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSReceiveOptionsCheckUtility.java

```diff
@@ -98,32 +84,25 @@ public <E extends ParserRuleContext> void checkOptions(E ctx) {
   }
 
   private void checkGroupOne(CICSParser.Cics_receive_group_oneContext ctx) {
-    checkHasMandatoryOptions(ctx.cics_length_flength(), ctx, "LENGTH or FLENGTH");
-    if (!ctx.ASIS().isEmpty() || !ctx.BUFFER().isEmpty()) {
-      checkHasIllegalOptions(ctx.LEAVEKB(), "LEAVEKB");
-    }
-  }
-
-  private void checkGroupTwo(CICSParser.Cics_receive_group_twoContext ctx) {
-    checkHasMandatoryOptions(ctx.cics_length_flength(), ctx, "LENGTH or FLENGTH");
-    checkHasMandatoryOptions(ctx.cics_into_set(), ctx, "INTO or SET");
-  }
-
-  private void checkGroupThree(CICSParser.Cics_receive_group_threeContext ctx) {
-    checkHasMandatoryOptions(ctx.cics_length_flength(), ctx, "LENGTH or FLENGTH");
+    checkHasMutuallyExclusiveOptions("INTO or SET", ctx.INTO(), ctx.SET());
+    if (!ctx.SET().isEmpty())
+      checkHasExactlyOneOption("LENGTH or FLENGTH", ctx, ctx.cics_length_flength());
+    checkHasMutuallyExclusiveOptions("MAXLENGTH or MAXFLENGTH", ctx.cics_maxlength());
```

Comment:
I think there should be checkCicsMaxlength function, that internally calls `checkHasMutuallyExclusiveOptions("...", ctx.MAXLENGTH(), ctx.MAXDLENGTH())`

Reply:
@chacebot, see how I did it here: https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/commit/1e19608440789537e56143f416f9681727575e87